src/app/game.ts

```diff
@@ -0,0 +1,373 @@
+import {
+  Building,
+  Unit,
+  House,
+  Wall,
+  Warehouse,
+  Goblin,
+  Knight,
+  Goldmine,
+} from './components'
+import {
+  Difficulty,
+  ResourceType,
+  Resource,
+  ONE_SECOND,
+  Fractions,
+} from './types'
+import { randomBetween } from './utils'
+
+const DEFAULT_GOLD = 100
+const DEFAULT_WOOD = 100
+const DEFAULT_STONE = 100
+const DEFAULT_STORAGE_CAPACITY = 600
+
+const START_POPULATION = 100
```

Comment:
Maybe we can move some things of them to contants folder (not sure if it is a good idea)